### DISCUSSION TITLE
I'm at a loss here using it with HomeAssisatant

### DISCUSSION THREAD

**Discussion by DerDet** [squishykid/solax#92] [Q&A] [answered]
Hi,
I'm at a loss here. I want to use this wonderful Plugin for Homassistant.
But the Solax Integration in HA is not the same as shown in the project.
I use HA (HA-OS) and I'm not able to install via Python script.
so, is there anybody out there, who can tell me, if I can use it into HA?

Thanks a lot
Best
Der Det

**Comment by VadimKraus** [comment, marked as answer]
Hey  @DerDet ,

The library just contains the bare minimum to connect and retrieve data from a variety of solax inverters, i.e. you can use it in different projects. 

https://www.home-assistant.io/integrations/solax/ is an HA integration which uses version 0.3.0 of this library.
 
This means the integration and the library are not the same. Since the HA integration requires a lot of additional stuff.
[https://github.com/home-assistant/core/tree/dev/homeassistant/components/solax](https://github.com/home-assistant/core/tree/dev/homeassistant/components/solax)

If it does not work via this integration, i.e. you receive a connection error, it might be the case that your inverter is not yet supported.
You might want to have a look at the logs, there might be a message referencing the [https://github.com/squishykid/solax/wiki/DiscoveryError](https://github.com/squishykid/solax/wiki/DiscoveryError)

I hope this helps

**Reply by DerDet** [reply]
Hey Vadim,
Thank you for your reply.
Your investigations on the inverter replies are great. I now use the RESTful API from HA to get the values. This works with the normal IP from the inverter.

**Comment by mapotn** [comment]
hi @DerDet have you solved your problem? I have same issue. thanks

**Reply by DerDet** [reply]
Hi @mapotn 
finaly, I descide not take deeper investigations into this. I resolved it by using a Rest Sensor in HA to get the data from my inverter.
It was helpfull to use the python scripts from the inverter folder to identify the correct fields.
If you need help, I can post my config.

**Reply by VadimKraus** [reply]
Why did you not create an issue with the response you get? Provide some meta data like what is your model etc?
Also if you already went through all the trouble of creating a mapping (for you personally), it should be quiet easy to include your model in the official addon...
